addons/udes_delivery_control/models/stock_picking.py

```diff
@@ -267,7 +360,7 @@ def action_delivery_control_done(self):
         for picking in self.filtered("u_is_delivery_control"):
             picking.write({"state": "done", "date_done": fields.Datetime.now()})
 
-            if not picking.u_goods_in_picking_id:
+            if not picking.u_goods_in_picking_id and picking.u_is_backload:
```

Comment:
This needs to be `picking.u_is_unload` not `picking.u_is_backload`.

We only want to generate a Goods In if the Delivery Control is unload or both unload and backload. If it is backload only then a Goods In shouldn't be generated.

Reply:
The tests were failing because of this :) 
Done